### DISCUSSION TITLE
Launching Foxglove Node with a premade layout

### DISCUSSION THREAD

**Discussion by julietcookies** [foxglove/community#789] [General]
To whom it may concern,

I'm wondering how I could launch Foxglove with my launch file containing a pre-made layout. Right now, I'm launching Foxglove in the following way:

```
def generate_launch_description():
        foxglove_bridge = Node(
                package='foxglove_bridge',
                executable='foxglove_bridge',
                name='foxglove_bridge',
                output='screen'
        )
        ld.add_action(foxglove_bridge)
        return ld
```
The Foxglove connection has been established and when I open the Foxglove Web application I see all of the relevant ROS2 topics being communicated. 

The problem is, that I can only import my panel layout ("layout.jason") after I open the Foxglove Web application using the import button on the UI. I would like my launch file to make it so that once I open the Web application it already imports my costume panel and there is no need to click any import buttons to get it to do so. 

I tried adding the following to my code (under the foxglove_bridge node):
`        parameters=[{'config_file': config_file_path}]`
where the config_file_path was the path to "layout.jason" but it still didn't work.

Can you help be by letting me know how I would do this?

Thanks in advance!

**Comment by achim-k** [comment]
You have to configure the layout in Foxglove Studio (the Foxglove web application) and not `foxglove_bridge`. The latter is only responsible for bridging ROS topics etc. to Foxglove Studio, but the way these are displayed is configured only in Foxglove Studio. Take a look at https://foxglove.dev/docs/studio/layouts for more details on how to configure a Studio Layout

**Reply by julietcookies** [reply]
Thank you for your reply!

I know how to configure using the web application and also how to save the layout as a json file on my computer and later upload it through the web application. Nonetheless, I would like my launch file to launch the application with a specific layout. I want someone to be able to just run my launch file through the terminal and have the web application working with a specific layout. Is that possible? If so, how?

Thanks again!

**Reply by achim-k** [reply]
You can do that by providing a specific link that contains the datasource URL and the layout ID. See https://foxglove.dev/docs/studio/building-and-sharing-links for details.

**Comment by jtbandes** [comment]
You could also explore using the Docker image to do this, it has an option to override the default layout: https://github.com/foxglove/studio#overriding-the-default-layout  But note that the Docker image contains only the open-source code and so it will have some features missing compared to the version at studio.foxglove.dev / in the desktop app.

**Reply by peterdavidfagan** [reply]
This has been super useful, Is it feasible to load more than one layout? 

I have a controller debug dashboard but I also want to have a dashboard for cameras + motion planning.

**Reply by amacneil** [reply]
The multi-layout code isn't open source, but the most common solution is to create a single layout with multiple tabs.

**Reply by peterdavidfagan** [reply]
Thanks @amacneil I'll follow the above pattern.
